plugins/integrations/kubernetes-service/src/main/java/com/cloud/kubernetes/cluster/actionworkers/KubernetesClusterResourceModifierActionWorker.java

```diff
@@ -537,6 +521,7 @@ protected void removePortForwardingRules(final IpAddress publicIp, final Network
                 for (PortForwardingRuleVO pfRule : pfRules) {
                     if (pfRule.getVirtualMachineId() == vmId) {
                         portForwardingRulesDao.remove(pfRule.getId());
+                        logger.debug(String.format("The Port forwarding rule [%s] with the id [%s] was removed.", pfRule.getName(), pfRule.getId()));
```

Comment:
No need to use `String.format()` here, with the updated log4j.

Reply:
see https://github.com/apache/cloudstack/pull/8801#discussion_r1639482249